website/docs/tutorials/08-cairo-1-support/01-testing.md

```diff
@@ -0,0 +1,76 @@
+---
+sidebar_label: Testing
+---
+
+# Running tests with Cairo 1
+
+:::info
+This functionality is in alpha stage, expect rapid iteration
+:::
+
+
+## Writing your first test
+
+To make a test function, you need to mark the function with a decorator `#[test]`
+
+```
+#[test]
+fn test_bool_operators() {
+    assert(1 == 1);
+}
+```
+
+To write a test which fails, you will need to use `panic`, here's how you do it:
+
+```
+use array::ArrayTrait;
+
+// Single value in panic payload
+#[test]
+fn test_panic_single_value() {
+    let mut data = array_new::<felt>();
+    array_append::<felt>(ref data, 21);
+    panic(data)
+}
+
+// Multiple values in panic payload
+#[test]
+fn test_panic_multiple_values() {
+    let mut data = array_new::<felt>();
+    array_append::<felt>(ref data, 101);
+    array_append::<felt>(ref data, 102);
+    array_append::<felt>(ref data, 103);
+    panic(data)
+}
+```
+
+## Running the tests
+
+To run cairo 1 tests, there is a special command called `test-cairo1`.
+It is a sister-command to `test` command, it will collect all the tests in the given directory/module, run them, and print out a summary.
+
+Tests are ran on Cairo VM, so no Starknet syscalls are available from the test code.
+
+:::info
+There is no support currently for starknet contracts at the moment, it's a work in progress.
+:::
+
+## Caveats
+### 1. Test collecting
+`test-cairo1` will collect all tests ending with `.cairo`, since there's no distinction between cairo 0 and cairo 1 files in terms of extension right now.
+
+That means that you will either have to specify a regex to match your test names (see [command reference](../cli-reference.md#test-cairo1)), or keep them in a separate directory to avoid syntax errors.
+
+A `test_` file prefix or `_test` postfix is required as well, to mark the files as test suites.
+
+### 2. Test state
+
+`__setup__` and `<test_name>_setup` from previous version are not supported for now.
+
+### 3. Test function type
+
+A test function must not return any values, and be panickable for correct test result assessment
+
+If your last statement returns a value, you must follow it with a line with blank statement (`;`).
```

Comment:
This is a bit confusing. I assume you it's about this syntax that if you miss the trailing `;` for the last command it is a return value for the function. If so, I'd rearrange this sentence a bit:
```
Please note, that cairo1 uses the rust-like syntax for the return values, so if the last instruction in a function doesn't have a ';' at the end, cairo1 treats this instruction as a returning one. That may lead to errors.
```

I don't know, I guess my suggested line isn't perfect but this is what I have in my mind more or less.

Reply:
I added an example